File: flutter/lib/screens/detail_screen.dart
import 'package:flutter/material.dart';
import 'package:multi_notes/models/models.dart';
import 'package:multi_notes/services/notes_service.dart';

class DetailScreen extends StatelessWidget {
  final Note note;
  final int index;
  const DetailScreen({Key? key, required this.note, required this.index})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(note.title),
      ),
      body: Container(
        padding: EdgeInsets.all(16),
        child: Column(
          mainAxisAlignment: MainAxisAlignment.start,
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              "Description",
              style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
            ),
            SizedBox(
              height: 10,
            ),
            Text(
              note.description,
              style: TextStyle(fontSize: 16),
            ),
            SizedBox(
              height: 10,
            ),
            Container(
              width: double.maxFinite,
              child: OutlinedButton(
                style: OutlinedButton.styleFrom(primary: Colors.red),
                onPressed: () {
                  openDeleteDialog(context);
                },
                child: Text("REMOVE NOTE"),
              ),
            ),
          ],
        ),
      ),
    );
  }

  void openDeleteDialog(BuildContext context) {
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: Text("Delete note"),
        content: Text("Are you sure you would like to delete this note?"),
        actions: [
          TextButton(
            onPressed: () {
              notesService.deleteNote(index);
              Navigator.popUntil(context, (route) => route.isFirst);
            },
            child: Text("DELETE"),
          ),
          TextButton(
            onPressed: () {},
            child: Text("CANCEL"),
          ),
        ],
      ),
    );
  }
}

File: react_native/src/screens/DetailScreen.js
import {useFocusEffect} from '@react-navigation/native';
import React from 'react';
import {View, Text, StyleSheet, Alert} from 'react-native';
import {Button} from 'react-native-paper';
import {notesService} from '../shared/NotesService';

const DetailScreen = ({navigation, route}) => {
  useFocusEffect(
    React.useCallback(() => {
      navigation.setOptions({
        title: route.params.title,
      });
    }, []),
  );

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Description:</Text>
      <Text style={styles.description}>{route.params.description}</Text>
      <Button
        color="red"
        mode="outlined"
        onPress={() => {
          Alert.alert(
            'Delete note',
            'Are you sure you would like to delete this note?',
            [
              {
                text: 'Delete',
                style: 'default',
                onPress: () => {
                  notesService.removeNote(route.params.index);
                  navigation.goBack();
                },
              },
              {
                text: 'Cancel',
                style: 'cancel',
                onPress: () => {},
              },
            ],
          );
        }}>
        Remove Note
      </Button>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
  },
  title: {
    fontSize: 19,
    fontWeight: '500',
    marginBottom: 5,
  },
  description: {
    fontSize: 16,
    marginBottom: 15,
  },
});

export default DetailScreen;
